dotnet/src/SemanticKernel.Core/Orchestration/KernelExecutionContextExtensions.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Extension methods for <see cref="IKernelExecutionContext"/>
+/// </summary>
+public static class KernelExecutionContextExtensions
```

Comment:
These RunAsync overloads -- if each one is not being used right now by something in the repo, please don't add them. I'd like to work toward removing "pipeline" and instead have kernel (and kernelContextExtensions) operate with `IPlan`.